client/pluginmanager/csimanager/volume.go

```diff
@@ -166,7 +166,7 @@ func (v *volumeManager) stageVolume(ctx context.Context, vol *structs.CSIVolume,
 	// CSI NodeStageVolume errors for timeout, codes.Unavailable and
 	// codes.ResourceExhausted are retried; all other errors are fatal.
 	return v.plugin.NodeStageVolume(ctx,
-		vol.ID,
+		vol.ExternalID,
```

Comment:
There's a getter method `vol.RemoteID()` for this. Let's use that here for consistency with the `v.plugin.NodePublishVolume` call.